network/src/gossip.rs

```diff
@@ -461,6 +482,27 @@ impl MessageValidationData {
 			Err(())
 		}
 	}
+
+	fn check_chunk(&self, msg: &ErasureChunkMessage) -> Result<(), ()> {
```

Comment:
this function IMO is not fulfilling the requirements of a DoS-proof gossip.
we have to account for malicious validators sending us chunks which don't align to anything.

The amount of candidates is bounded by the number of validators and by the slashing condition on `Candidate` message equivocations. We should only accept erasure chunks for candidates we know. And in that case, I don't see why it needs to be signed by a validator, since it can be checked against the root in the `CandidateReceipt`.